src/mprk.jl

```diff
@@ -664,39 +605,46 @@ function perform_step!(integrator, cache::MPRK22Cache, repeat_step = false)
     @unpack tmp, atmp, P, P2, D, D2, M, σ, thread, weight = cache
     @unpack a21, b1, b2, c2, small_constant = cache.tab
 
-    uprev .= uprev .+ small_constant
-
     f.p(P, uprev, p, t) # evaluate production terms
-    sum_destruction_terms!(D, P) # store destruction terms in D
     integrator.stats.nf += 1
+    @.. broadcast=false P2=a21 * P
 
-    build_mprk_matrix!(M, a21, P, D, uprev, dt)
-    # Same as linres = M \ uprev
+    # avoid division by zero due to zero Patankar weights
+    @.. broadcast=false σ=uprev + small_constant
+
+    build_mprk_matrix!(P2, P2, σ, dt)
```

Comment:
This is safe for fully conservative systems but may require some thought in general. Would it make sense to still use another matrix `M` here to collect the terms? If not, can we remove `M` from the `cache`?

Reply:
I think we can remove `M`, since we can always use the "last P-matrix" to store the matrix of the linear system.

Reply:
Could we maybe use `P` and `M` in this case to clarify the meaning of variables? We compute the production terms in `P` and build the MPRK matrix in `M` (or something like this instead of `P1` and `P2`)?

Reply:
Ok, let's do it like you suggest 👍

Reply:
I just found that it was already implemented this way for `MPE()`.